### PR TITLE
Issue735/bailfilter

### PR DESCRIPTION
**Pull request by ibixina**
Bail cost for jail filter #735 

added a bail cost filter in the jail filter
modified the existing code in ttJailFilter.js to have a new textbox for bail cost filter.

image

### CODE REVIEW

**Comment by ibixina** [comment]
This pull request fixes #735 .

File: extension/scripts/features/jail-filter/ttJailFilter.js [hunk 1]
```diff
@@ -108,6 +108,13 @@
 		filterContent.appendChild(scoreFilter.element);
 		localFilters["Score Filter"] = { getStartEnd: scoreFilter.getStartEnd, updateCounter: scoreFilter.updateCounter };
 
+		const bailCostFilter = createTextbox({ type: "number", description: "Bail Cost less than", min: 0 });
+		bailCostFilter.setValue(filters.jail.bailCost || 9999999);
```
Comment (DeKleineKobini) [inline extension/scripts/features/jail-filter/ttJailFilter.js:112]: Seems like a bad default to be honest, as bail cost might be higher than this. Safest would probably just keep it empty as default and have that disable the filter.

File: extension/scripts/features/jail-filter/ttJailFilter.js [hunk 2]
```diff
@@ -203,8 +227,11 @@
 			}
 
 			// Time
-			const timeMatch = li.find(".info-wrap .time").textContent.match(JAIL_FILTER_TIME_REGEX);
-			const timeLeftHrs = timeMatch ? parseInt(timeMatch[0]) : 0;
+			const timeMatch = li.find(".info-wrap .time").textContent;
```
Comment (DeKleineKobini) [inline extension/scripts/features/jail-filter/ttJailFilter.js:230]: No longer an accurate name, should be renamed..

**Review by DeKleineKobini — CHANGES_REQUESTED** [review]
From your screenshot, it seems so packed with this added, so I'm requesting a review from the other collaborator as well.

As mentioned in the [contributing guide](https://github.com/Mephiles/torntools_extension/blob/master/CONTRIBUTING.md), you are missing the changelog entry and you aren't in the `team.js` file yet either.

**Comment by ibixina** [comment]
do you think it might be better if it was under the faction filter?

**Comment by Sashank999** [comment]
> do you think it might be better if it was under the faction filter?

Rather than grouping up filters, I have added code for making the filters wrap. Currently, only Auction House Filter and this new Jail Filter will be affected by this change.

File: extension/scripts/features/jail-filter/ttJailFilter.js [hunk 3]
```diff
@@ -108,6 +108,13 @@
 		filterContent.appendChild(scoreFilter.element);
 		localFilters["Score Filter"] = { getStartEnd: scoreFilter.getStartEnd, updateCounter: scoreFilter.updateCounter };
 
+		const bailCostFilter = createTextbox({ type: "number", description: "Bail Cost less than", min: 0 });
```
Comment (Sashank999) [inline extension/scripts/features/jail-filter/ttJailFilter.js:111]: I suggest the description as "Maximum Bail Cost". It is lesser verbose IMO.

**Review by Sashank999 — CHANGES_REQUESTED** [review]
That is the change that I suggest. Everything else is good to go.

File: extension/scripts/features/jail-filter/ttJailFilter.js (hunk 3 above)
Comment (ibixina) [reply extension/scripts/features/jail-filter/ttJailFilter.js]: made the changes

**Pull request merged**